src/api/Controllers/UploadController.cs

```diff
@@ -140,8 +141,16 @@ public BoreholeMap()
             };
 
             AutoMap(config);
-            Map(b => b.LocationX).Name("location_x_lv_95");
-            Map(b => b.LocationY).Name("location_y_lv_95");
+            Map(b => b.OriginalReferenceSystem).Convert(args =>
+            {
+                var locationX = args.Row.GetField<double?>("location_x");
+                var locationY = args.Row.GetField<double?>("location_y");
+                return locationX == null || locationY == null ? null : locationX >= 2_000_000 ? ReferenceSystem.LV95 : ReferenceSystem.LV03;
+            });
+            Map(b => b.LocationX).Convert(args => { return args.Row.GetField<double?>("location_x") >= 2_000_000 ? args.Row.GetField<double?>("location_x") : null; });
```

Comment:
PP: Ohne die geschweiften Klammern lässt sich das einfacher ausdrücken:
```suggestion
            Map(b => b.LocationX).Convert(args => args.Row.GetField<double?>("location_x") >= 2_000_000 ? args.Row.GetField<double?>("location_x") : null);
```